internal/controller/pod_controller.go

```diff
@@ -0,0 +1,95 @@
+// Package controller provides controllers for managing ModelValidation resources
+package controller
+
+import (
+	"context"
+
+	"github.com/sigstore/model-validation-operator/internal/constants"
+	"github.com/sigstore/model-validation-operator/internal/tracker"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// PodReconciler reconciles Pod objects to track injected pods
+type PodReconciler struct {
+	client.Client
+	Scheme  *runtime.Scheme
+	Tracker tracker.StatusTracker
+}
+
+// +kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch;update
+// +kubebuilder:rbac:groups="",resources=pods/finalizers,verbs=update
+// +kubebuilder:rbac:groups=ml.sigstore.dev,resources=modelvalidations,verbs=get;list
+// +kubebuilder:rbac:groups=ml.sigstore.dev,resources=modelvalidations/status,verbs=update
+
+// Reconcile handles pod events to update ModelValidation status
+func (r *PodReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	logger := log.FromContext(ctx)
+
+	pod := &corev1.Pod{}
+	if err := r.Get(ctx, req.NamespacedName, pod); err != nil {
+		if errors.IsNotFound(err) {
+			logger.V(1).Info("Pod deleted, removing from tracking", "pod", req.NamespacedName)
+			if err := r.Tracker.RemovePodByName(ctx, req.NamespacedName); err != nil {
+				logger.Error(err, "Failed to remove deleted pod from tracking", "pod", req.NamespacedName)
+				return reconcile.Result{}, err
+			}
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	if !pod.DeletionTimestamp.IsZero() {
+		logger.Info("Handling pod deletion", "pod", req.NamespacedName)
+
+		if err := r.Tracker.RemovePodEvent(ctx, pod.UID); err != nil {
+			logger.Error(err, "Failed to remove pod from tracking")
+			return reconcile.Result{}, err
+		}
+
+		var finalizers []string
```

Comment:
For managing finalizers, it's best practice to use the dedicated helper functions from [controller-runtime/controllerutil](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/controller/controllerutil).

Instead of manually manipulating the finalizer slice, please use:
- `controllerutil.AddFinalizer`
- `controllerutil.RemoveFinalizer`
- `controllerutil.ContainsFinalizer`

Reply:
No problem, I can switch to those.

Reply:
@osmman I've pushed the [switch to the controller utils helpers](https://github.com/sigstore/model-validation-operator/compare/719b1b6a797c9bb8c168c67f4a3b6ae60f48ae6d..83a783d713a0ea7574a70af8581aabb435146be1), please take a look